### ISSUE TITLE
Users with collections "edit" permissions and no data access permissions can't edit question metadata

### ISSUE THREAD

**Issue by tlrobinson** [metabase/metabase#11719]
**Describe the bug**
A user with no "data access" permissions but "edit" collection permissions should be able to view and edit the question metadata (`name`, `description`, `collection_id`, and `archived`) but not the question definition (unless they are allowed to run the resulting ad-hoc question?). Currently they can view the question but not edit anything about the question.

**Logs**

```
01-14 23:56:07 WARN middleware.log :: PUT /api/card/40 403 41.5 ms (33 DB calls)
"You don't have permissions to do that."
```

**To Reproduce**
1. Create a non-admin user and revoke their data access permissions from "All Users"
2. Have an admin save a question to a collection the non-admin user does have "edit" collection permissions for
4. Have the non-admin user view the question, which should succeed. Try having them edit the question `name` or `description` (via "Edit this question" entity menu), `collection_id` (via "Move" entity menu item), or `archived` (via "Archive" entity menu item), but you will get a permissions error.

**Expected behavior**
The user should be able to edit the question metadata, but not the question definition (because they have no data permissions)

**Screenshots**

**Information about your Metabase Installation:**

You're on version v0.35.0-snapshot
Built on 2020-01-13
Hash: 2807015
Branch: master

**Severity**

P2?

**Comment by camsaul** [comment]
You actually can edit the metadata right now as long as you don't pass in `dataset_query`. I'm working on a fix tho so it lets you pass in `dataset_query` as long as it's unchanged from its current value

**Comment by camsaul** [comment]
Fixed by #15424
